### PR TITLE
En 8334 separate list for seeders

### PR DESCRIPTION
**Pull request by iulianpascalau**
- implemented a dedicated list for seeders as to not consider them as unknown peers
- added argument structs on some constructors
- p2p package general cleanup

### CODE REVIEW

**Review by sasurobert — APPROVED** [review]

File: p2p/libp2p/networksharding/listsSharder.go [hunk 1]
```diff
@@ -16,18 +17,19 @@ import (
...
 const minAllowedValidators = 1
 const minAllowedObservers = 1
 const minUnknownPeers = 1
 
 const intraShardValidators = 0
-const intraShardObservers = 1
-const crossShardValidators = 2
-const crossShardObservers = 3
-const unknown = 4
+const intraShardObservers = 10
```
Comment (raduchis) [inline p2p/libp2p/networksharding/listsSharder.go:28]: these are used like enums, right?

File: sharding/networksharding/peerShardMapper.go [hunk 2]
```diff
@@ -31,54 +32,56 @@ var _ p2p.PeerShardResolver = (*PeerShardMapper)(nil)
...
 	nodesCoordinator sharding.NodesCoordinator
 }
 
+// ArgPeerShardMapper is the initialization structure for the PeerShardMapper implementation
+type ArgPeerShardMapper struct {
+	PeerIdPk         storage.Cacher
+	FallbackPkShard  storage.Cacher
+	FallbackPidShard storage.Cacher
+	NodesCoordinator sharding.NodesCoordinator
+	EpochStart       uint32
```
Comment (raduchis) [inline sharding/networksharding/peerShardMapper.go:52]: StartEpoch is maybe better?

File: factory/processComponents.go [hunk 3]
```diff
@@ -1236,13 +1236,14 @@ func createNetworkShardingCollector(
...
 
-	psm, err := networksharding.NewPeerShardMapper(
-		cachePkPid,
-		cachePkShardID,
-		cachePidShardID,
-		nodesCoordinator,
-		epochStart,
-	)
+	arg := networksharding.ArgPeerShardMapper{
+		PeerIdPk:         cachePkPid,
```
Comment (raduchis) [inline factory/processComponents.go:1240]: I didnt' think that the parameters starting with cache are actually caches until I looked in the code. Could you please rename them to: PkPidCache etc? Also for the ArgPeerShardMapper PeerIdPk, FallbackPkShard,FallbackPidShard I thought they are Pk -> byte[], Shard -> uint while all of them are caches. Are they actually Fallbacks?

**Review by raduchis — COMMENTED** [review]
some renaming might be needed

File: p2p/libp2p/networksharding/listsSharder.go (hunk 1 above)
Comment (iulianpascalau) [reply p2p/libp2p/networksharding/listsSharder.go:28]: Yes, as to easily add the next ones (FullHistoryObservers, for example)

File: sharding/networksharding/peerShardMapper.go (hunk 2 above)
Comment (iulianpascalau) [reply sharding/networksharding/peerShardMapper.go:52]: done

File: factory/processComponents.go (hunk 3 above)
Comment (iulianpascalau) [reply factory/processComponents.go:1240]: Ok, will rename those.
The fallback caches are mainly used for observers or for validators if the NodesCoordinator fails to provide the correct mapping between Pk and ShardID. That is why I called them fallback.

Comment (iulianpascalau) [reply factory/processComponents.go:1240]: done

File: p2p/libp2p/networksharding/listsSharder.go (hunk 1 above)
Comment (raduchis) [reply p2p/libp2p/networksharding/listsSharder.go:28]: ok

File: sharding/networksharding/peerShardMapper.go [hunk 4]
```diff
@@ -58,30 +59,30 @@ func NewPeerShardMapper(arg ArgPeerShardMapper) (*PeerShardMapper, error) {
 	if check.IfNil(arg.NodesCoordinator) {
 		return nil, sharding.ErrNilNodesCoordinator
 	}
-	if check.IfNil(arg.PeerIdPk) {
-		return nil, sharding.ErrNilCacher
+	if check.IfNil(arg.PeerIdPkCache) {
+		return nil, fmt.Errorf("%w for PeerIdPkCache", sharding.ErrNilCacher)
```
Comment (raduchis) [inline sharding/networksharding/peerShardMapper.go:63]: :+1:

**Review by raduchis — APPROVED** [review]

File: factory/processComponents.go (hunk 3 above)
Comment (raduchis) [reply factory/processComponents.go]: ok

File: sharding/networksharding/peerShardMapper.go (hunk 2 above)
Comment (raduchis) [reply sharding/networksharding/peerShardMapper.go]: ok

**Review by sasurobert — APPROVED** [review]

**Pull request merged**